akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala

```diff
@@ -272,6 +272,20 @@ class SourceSpec extends AkkaSpec with DefaultTimeout {
     }
   }
 
+  "Cycle Source" must {
+
+    "continuously generate the same sequence" in {
+      val expected = Seq(1, 2, 3, 1, 2, 3, 1, 2, 3)
+      Source.cycle(1, 2, 3).grouped(9).runWith(Sink.head).futureValue should ===(expected)
+    }
+
+    "reject creation from empty sequence" in {
+      intercept[IllegalArgumentException] {
+        Source.cycle()
```

Comment:
Let's make this not compile instead of throwing.
Signature should be:

```scala
def cycle[T](first: T, rest: T*): Soruce[T, NotUsed] = ???
// cycle(1,2,3) // ok
// cycle(1) // ok
// repeat(1) should be implemented in terms of cycle(1) I guess?
// cycle() // won't compile
```